File: Python/Rotate List.py
class Solution:
    # @param head, a ListNode
    # @param k, an integer
    # @return a ListNode
    def rotateRight(self, head, k):
        if k == 0:
            return head
        if not head:
            return head

        rotate = head
        length = 1
        while rotate.next:
            rotate = rotate.next
            length += 1
        
        rotate.next = head
        k = k % length
        
        for i in range(length - k):
            rotate = rotate.next
        head = rotate.next
        rotate.next = None
        
        return head
        

File: Java/src/Rotate List.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 * int val;
 * ListNode next;
 * ListNode(int x) {
 * val = x;
 * next = null;
 * }
 * }
 */
public class Solution
{
  public ListNode rotateRight(ListNode head, int n)
  {
    if (n == 0 || head == null)
      return head;
    ListNode rotate = head;
    int length = 1;

    while (rotate.next != null)
    {
      rotate = rotate.next;
      length++;
    }

    rotate.next = head;
    int k = n % length;

    for (int i = 0; i < length - k; i++)
      rotate = rotate.next;
    head = rotate.next;
    rotate.next = null;
    return head;

  }
}
